analysis/estimate/notebooks/run-estimate.Rmd

```diff
@@ -0,0 +1,289 @@
+
+
+immune_df <- expr_wide_df |>
+  dplyr::filter(score_type == "immune")
+summary(lm(bulk_tpm ~ 1 + offset(bulk_counts), data = immune_df))
+
+```
+
+Indeed, effect sizes are not very large in the scheme of the full range of values, but they are highly significant: Compared to bulk counts, immune scores are relatively higher when calculated with TPM, and stromal scores are relatively lower whem calculated with TPM.
+
+Let's directly compare bulk to and pseudobulk.
+Since TPM and bulk counts were very similar, we'll just visualize with the bulk _counts_ (not TPM) here:
+
+
+```{r fig.width=10}
+compare_modalities_df <- estimate_df |>
+  tidyr::pivot_wider(
+    names_from = expression_type, 
+    values_from = score
+  ) 
+
+ggplot(compare_modalities_df) + 
+  aes(x = bulk_counts, y = pseudobulk) + 
+  geom_point() +
+  geom_abline(color = "darkorange2") +
+  geom_smooth(method = "lm") +
+  facet_grid(
+    rows = vars(score_type), 
+    cols = vars(project_id)
+  ) +
+  labs(title ="pseudobulk ~ bulk counts") +
+  theme(aspect.ratio = 1)
+```
+
+* There does not appear to be a significant relationship between modalities for SCPCP000002 and SCPCP000006, for either score 
+* For SCPCP000001, immune scores tend to be higher in pseudobulk but stromal scores are about the same between modalities
+* For SCPCP000006 and SCPCP0000017, immune scores tend to be lower in pseudobulk but stromal scores are about the same between modalities
+* Note that projects SCPCP000001 and SCPCP000002 are single-cell, but the other three projects are single-nucleus
+
+
+### SCPCP000001
+
+In SCPCP000001, we seem to have have a trimodal distribution of pseudobulk immune scores.
+Does any metadata variable separate these points?
+
+
+```{r}
+library_project1 <- library_metadata_df |> 
+  dplyr::filter(
+    scpca_project_id == "SCPCP000001", 
+    seq_unit == "bulk"
+  ) |> 
+  dplyr::select(scpca_sample_id, technology, library_method)
+
+project1_df <- sample_metadata_df |>
+  dplyr::filter(scpca_project_id == "SCPCP000001") |>
+  dplyr::inner_join(library_project1) |>
+  dplyr::inner_join(compare_modalities_df, by = c("scpca_sample_id" = "sample_id")) |>
+  dplyr::filter(score_type == "immune")
+```
+
+We'll first look at diagnosis:
+
+```{r}
+ggplot(project1_df) + 
+  aes(x = diagnosis, y = pseudobulk, color = subdiagnosis) + 
+  geom_jitter(width = 0.1) + 
+  ggtitle("pseudobulk ~ diagnosis") + 
+  theme(axis.text.x = element_text(angle = 30, hjust = 1))
+```
+
+The sample size here is too small to really draw a conclusion, but it's possible that the "epithelioid" smples are coming out with the highest immune scores.
+
+Let's look at a few other variables:
+
+
+```{r fig.height=10, fig.width=10}
+p1 <- ggplot(project1_df) + 
+  aes(x = age, y = pseudobulk) + 
+  geom_point() +
+  ggtitle("pseudobulk ~ age")
+
+p2 <- ggplot(project1_df) + 
+  aes(x = technology, y = pseudobulk) + 
+  geom_jitter(width = 0.1) + 
+  ggtitle("pseudobulk ~ technology")
```

Comment:
I don't know that with pseudobulk looking at single-end and paired-end makes a whole lot of sense? That really shouldn't affect the pseudobulk samples at all since that information is only relevant for bulk.

Reply:
Ah yes! definitely correct.